src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,27 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    public function resetExpiringDate()
+    {
+        $this->expiringDate = null;
+    }
+
+    /**
+     * @param \DateTimeImmutable $expiringDate
+     */
+    public function reduceExpiringDate(\DateTimeImmutable $expiringDate)
+    {
+        if (null === $this->expiringDate || $this->expiringDate > $expiringDate) {
```

Comment:
The method name is not clear to me. `reduce` means that it's going to expire earlier than before, right. But you say that the purpose is to "move forward".